src/backend/utils/misc/guc.c

```diff
@@ -8200,7 +8200,7 @@ static struct config_string ConfigureNamesString[] =
 			NULL
 		},
 		&standby_addr_host,
-		"localhost", NULL, NULL
+		"none", NULL, NULL
```

Comment:
No, segment rm process does not care what are saved in python code. So, it only knows there is a standby host named "none".  So deleting from hawq_dict does not help.